score.py

```diff
@@ -128,6 +134,75 @@ def run_inference(images_with_header: List[ImageWithHeader],
+    Return a list of DICOM series folders.
+
+    :param zip_file_path: Path to zip file.
+    :param model_folder: Path to model folder.
+    :return: List of DICOM series folders.
+    """
+    dicom_folders: List[Path] = []
+    extraction_folder = model_folder / "temp_extraction"
+    with zipfile.ZipFile(zip_file_path, 'r') as zip_file:
+        for zipped_file in zip_file.namelist():
+            if zipped_file.endswith('.dcm'):
+                zip_file.extract(zipped_file, str(extraction_folder))
+                extracted_file_path = extraction_folder / zipped_file
+                extracted_file_path_folder = extracted_file_path.parent
+                if extracted_file_path_folder not in dicom_folders:
+                    dicom_folders.append(extracted_file_path_folder)
+    return dicom_folders
+
+
+def convert_zipped_dicom_to_nifti(test_images: List[Path], model_folder: Path) -> Tuple[List[Path], List[Path]]:
+    """
+    Given a zip file, extract DICOM series and convert to Nifti format.
+
+    :param test_images: A list containing a single zip file.
+    :param model_folder: Scratch folder to extract files into.
+    :return: Pair of list of Nifti files and their corresponding reference series.
+    """
+    # Only a single zip file is supported.
+    if len(test_images) != 1 or not zipfile.is_zipfile(test_images[0]):
+        raise ValueError("Supply exactly one zip file in args.images.")
+    reference_series_folders = extract_zipped_dicom_series(test_images[0], model_folder)
+    converted_images: List[Path] = []
+    for i, reference_series_folder in enumerate(reference_series_folders):
+        nifti_filename = model_folder / f"temp_nifti_{i}.nii.gz"
+        load_dicom_series(reference_series_folder, nifti_filename)
+        converted_images.append(nifti_filename)
+    return converted_images, reference_series_folders
+
+
+def convert_nifti_to_zipped_dicom_rt(nifti_file: Path, reference_series: Path,
+                                     config: SegmentationModelBase) -> Path:
+    """
+    Given a Nifti file and a reference DICON series, convert Nifti file to DICOM-RT file.
+
+    :param nifti_file: Path to Nifti file.
+    :param reference_series: Path to folder containing reference DICOM series.
+    :param config: Model config.
+    :return: Path to DICOM-RT file.
+    """
+    dicom_rt_file = nifti_file.with_suffix(".dcm")
+    (stdout, stderr) = rtconvert(
+        in_file=nifti_file,
+        reference_series=reference_series,
+        out_file=dicom_rt_file,
+        struct_names=config.ground_truth_ids_display_names,
+        struct_colors=config.colours,
+        fill_holes=config.fill_holes)
+    logging.debug("stdout: %s", stdout)
+    logging.debug("stderr: %s", stderr)
+    dicom_rt_zip_file = dicom_rt_file.with_suffix(dicom_rt_file.suffix + '.zip')
+    with zipfile.ZipFile(dicom_rt_zip_file, 'w') as dicom_rt_zip:
+        dicom_rt_zip.write(dicom_rt_file)
```

Comment:
there is no need to zip up the RT file, I think. They are usually small.

Reply:
The zip is in fact, 266 bytes larger than the uncompressed dicom-rt file. Javier wanted it zipped though.